### PR TITLE
Logging hackathon - runners.

### PR DESCRIPTION
**Pull request by M6AI**
## Bug / Requirement Description
Clearly and concisely describe the problem.

## Solution description
Describe your code changes in detail for reviewers.

## Checklist:
- [ ] Test
- [ ] Example (both test_plan.py and .rst)
- [ ] Documentation (API)
- [ ] News fragment present for release notes
- [ ] MS info leakage check
- [ ] For new driver: driver index page
- [ ] For new assertion: ui/pdf/std renderers, documentation
- [ ] For new cmdline arg: documentation

### CODE REVIEW

**Review by kn-ms — DISMISSED** [review]

File: testplan/runners/local.py
Comment (rnemes) [inline testplan/runners/local.py:1]: At line 77, logger.exception is not in C-type string format, is that intentional?

File: testplan/runners/pools/connection.py
Comment (rnemes) [inline testplan/runners/pools/connection.py:1]: At line 70 and 76 logger.exception are not in C-type string format.

File: testplan/runners/pools/process.py [hunk 1]
```diff
@@ -84,10 +81,10 @@ def _write_syspath(self, sys_path=None):
             self.logger.debug("Written sys.path to file: %s", f.name)
             self._syspath_file = f.name
 
-    def pre_start(self):
+    def pre_start(self) -> None:
         self._write_syspath()
 
-    def starting(self):
+    def starting(self) -> None:
```
Comment (rnemes) [inline testplan/runners/pools/process.py:87]: Line 91, .debug is not in C-type string format.

File: testplan/runners/pools/base.py [hunk 2]
```diff
@@ -491,7 +473,7 @@ def _handle_taskpull_request(
                         continue
                     else:
                         if self._can_assign_task_to_worker(task, worker):
-                            self.logger.test_info(
+                            self.logger.user_info(
                                 "Scheduling {} to {}{}".format(
```
Comment (rnemes) [inline testplan/runners/pools/base.py:477]: This is not in C-type string format.
Alongside with line 430, 492, 537, 554, 585, etc...

**Comment by M6AI** [comment]
> See separate comments on files

Thanks Robi, thanks for catching these, I drilled into it and there are many places in addition not visible on the diff.

File: testplan/runners/local.py [hunk 3]
```diff
@@ -74,9 +70,7 @@ def _loop(self):
                         )
                         result.report.status_override = Status.ERROR
                         result.report.logger.exception(
-                            "Exception for {} on {} execution: {}".format(
-                                next_uid, self, exc
-                            )
+                            f"Exception for {next_uid} on {self} execution: {exc}"
```
Comment (rnemes) [inline testplan/runners/local.py]: Here

File: testplan/runners/pools/remote.py [hunk 4]
```diff
@@ -345,12 +309,12 @@ def __init__(
...
 
     @staticmethod
-    def _worker_setup_metadata(worker, request, response):
+    def _worker_setup_metadata(worker, request, response) -> None:
         worker.respond(
             response.make(Message.Metadata, data=worker.setup_metadata)
         )
 
-    def _add_workers(self):
+    def _add_workers(self) -> None:
```
[the remark this answers is not in the document]
Comment (rnemes) [reply testplan/runners/pools/remote.py:317]: 326 is still not in the expected format.

File: testplan/runners/pools/base.py [hunk 5]
```diff
@@ -445,15 +423,14 @@ def handle_request(self, request: Message):
                 worker.respond(response.make(Message.Stop))
         else:
             self.logger.error(
-                "Unknown request: {} {} {} {}".format(
-                    request, dir(request), request.cmd, request.data
-                )
+                f"Unknown request: {request} {dir(request)} {request.cmd}"
```
Comment (rnemes) [inline testplan/runners/pools/base.py:426]: Shouldn't this be in C-style as well?

**Review by rnemes — APPROVED** [review]

**Pull request merged**